// @dhis2/app-adapter/build/cjs/components/FatalErrorBoundary.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FatalErrorBoundary = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _d2I18n = _interopRequireDefault(require("@dhis2/d2-i18n"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _FatalErrorBoundary = _interopRequireDefault(require("./styles/FatalErrorBoundary.style"));

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

const translatedErrorHeading = _d2I18n.default.t('An error occurred in the DHIS2 application.');

const replaceNewlinesWithBreaks = text => text.split('\n').reduce((out, line, i) => [...out, line, /*#__PURE__*/_react.default.createElement("br", {
  key: i
})], []);

class FatalErrorBoundary extends _react.Component {
  constructor(props) {
    super(props);

    _defineProperty(this, "toggleTechInfoDrawer", () => {
      this.setState({
        drawerOpen: !this.state.drawerOpen
      });
    });

    this.state = {
      error: null,
      errorInfo: null,
      drawerOpen: false
    };
  }

  componentDidCatch(error, errorInfo) {
    this.setState({
      error,
      errorInfo
    });
  }

  render() {
    const {
      children
    } = this.props;

    if (this.state.error) {
      return /*#__PURE__*/_react.default.createElement("div", {
        className: "jsx-".concat(_FatalErrorBoundary.default.__hash) + " " + "mask"
      }, /*#__PURE__*/_react.default.createElement(_style.default, {
        id: _FatalErrorBoundary.default.__hash
      }, _FatalErrorBoundary.default), /*#__PURE__*/_react.default.createElement("div", {
        className: "jsx-".concat(_FatalErrorBoundary.default.__hash) + " " + "container"
      }, /*#__PURE__*/_react.default.createElement("div", {
        className: "jsx-".concat(_FatalErrorBoundary.default.__hash) + " " + "message"
      }, _d2I18n.default.t('Something went wrong')), /*#__PURE__*/_react.default.createElement("div", {
        onClick: () => window.location.reload(),
        className: "jsx-".concat(_FatalErrorBoundary.default.__hash) + " " + "link"
      }, _d2I18n.default.t('Refresh to try again')), /*#__PURE__*/_react.default.createElement("div", {
        onClick: this.toggleTechInfoDrawer,
        className: "jsx-".concat(_FatalErrorBoundary.default.__hash) + " " + "drawerToggle"
      }, this.state.drawerOpen ? _d2I18n.default.t('Hide technical details') : _d2I18n.default.t('Show technical details')), /*#__PURE__*/_react.default.createElement("div", {
        className: "jsx-".concat(_FatalErrorBoundary.default.__hash) + " " + ((this.state.drawerOpen ? 'drawerVisible' : 'drawerHidden') || "")
      }, /*#__PURE__*/_react.default.createElement("div", {
        className: "jsx-".concat(_FatalErrorBoundary.default.__hash) + " " + "errorIntro"
      }, translatedErrorHeading, /*#__PURE__*/_react.default.createElement("br", {
        className: "jsx-".concat(_FatalErrorBoundary.default.__hash)
      }), _d2I18n.default.t('The following information may be requested by technical support.')), /*#__PURE__*/_react.default.createElement("div", {
        className: "jsx-".concat(_FatalErrorBoundary.default.__hash) + " " + "errorDetails"
      }, [replaceNewlinesWithBreaks(this.state.error.stack + '\n---' + this.state.errorInfo.componentStack)]))));
    }

    return children;
  }

}

exports.FatalErrorBoundary = FatalErrorBoundary;
FatalErrorBoundary.propTypes = {
  children: _propTypes.default.node.isRequired
};
