// @dhis2/app-adapter/build/cjs/components/LoginModal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LoginModal = void 0;

var _ui = require("@dhis2/ui");

var _react = _interopRequireWildcard(require("react"));

var _locales = _interopRequireDefault(require("../locales"));

var _api = require("../utils/api");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const staticUrl = process.env.REACT_APP_DHIS2_BASE_URL;

const LoginModal = () => {
  const [server, setServer] = (0, _react.useState)(staticUrl || window.localStorage.DHIS2_BASE_URL || '');
  const [username, setUsername] = (0, _react.useState)('');
  const [password, setPassword] = (0, _react.useState)('');
  const [isDirty, setIsDirty] = (0, _react.useState)(false);

  const isValid = val => val && val.length >= 2;

  const onSubmit = async e => {
    e.preventDefault();
    setIsDirty(true);

    if (isValid(server) && isValid(username) && isValid(password)) {
      window.localStorage.DHIS2_BASE_URL = server;

      try {
        await (0, _api.post)("".concat(server, "/dhis-web-commons-security/login.action"), "j_username=".concat(encodeURIComponent(username), "&j_password=").concat(encodeURIComponent(password)));
      } catch (e) {
        console.log('TODO: This will always error and cancel the request until we get a real login endpoint!');
      } // TODO: Hacky solution... this shouldn't require a reload


      window.location.reload();
    }
  };

  return /*#__PURE__*/_react.default.createElement(_ui.Modal, {
    open: true,
    small: true,
    dataTest: "dhis2-adapter-loginmodal"
  }, /*#__PURE__*/_react.default.createElement("form", {
    onSubmit: onSubmit
  }, /*#__PURE__*/_react.default.createElement(_ui.ModalTitle, null, _locales.default.t('Please sign in')), /*#__PURE__*/_react.default.createElement(_ui.ModalContent, null, !staticUrl && /*#__PURE__*/_react.default.createElement(_ui.InputField, {
    dataTest: "dhis2-adapter-loginserver",
    error: isDirty && !isValid(server),
    label: _locales.default.t('Server'),
    name: "server",
    type: "text",
    value: server,
    onChange: input => setServer(input.value)
  }), /*#__PURE__*/_react.default.createElement(_ui.InputField, {
    dataTest: "dhis2-adapter-loginname",
    error: isDirty && !isValid(username),
    label: _locales.default.t('Username'),
    name: "j_username",
    type: "text",
    value: username,
    onChange: input => setUsername(input.value)
  }), /*#__PURE__*/_react.default.createElement(_ui.InputField, {
    dataTest: "dhis2-adapter-loginpassword",
    error: isDirty && !isValid(password),
    label: _locales.default.t('Password'),
    name: "j_password",
    type: "password",
    value: password,
    onChange: input => setPassword(input.value)
  })), /*#__PURE__*/_react.default.createElement(_ui.ModalActions, null, /*#__PURE__*/_react.default.createElement(_ui.Button, {
    primary: true,
    dataTest: "dhis2-adapter-loginsubmit",
    type: "submit"
  }, _locales.default.t('Sign in')))));
};

exports.LoginModal = LoginModal;
