// @dhis2/app-adapter/build/es/index.js
import _JSXStyle from "styled-jsx/style";
import { HeaderBar } from '@dhis2/ui';
import PropTypes from 'prop-types';
import React from 'react';
import { Alerts } from './components/Alerts';
import { AuthBoundary } from './components/AuthBoundary';
import { FatalErrorBoundary } from './components/FatalErrorBoundary';
import { ServerVersionProvider } from './components/ServerVersionProvider';
import { styles } from './styles.js';

const App = ({
  url,
  apiVersion,
  appName,
  children
}) => /*#__PURE__*/React.createElement(FatalErrorBoundary, null, /*#__PURE__*/React.createElement(ServerVersionProvider, {
  url: url,
  apiVersion: apiVersion
}, /*#__PURE__*/React.createElement("div", {
  className: "jsx-".concat(styles.__hash) + " " + "app-shell-adapter"
}, /*#__PURE__*/React.createElement(_JSXStyle, {
  id: styles.__hash
}, styles), /*#__PURE__*/React.createElement(HeaderBar, {
  appName: appName
}), /*#__PURE__*/React.createElement(AuthBoundary, {
  url: url
}, /*#__PURE__*/React.createElement("div", {
  className: "jsx-".concat(styles.__hash) + " " + "app-shell-app"
}, children)), /*#__PURE__*/React.createElement(Alerts, null))));

App.propTypes = {
  appName: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  apiVersion: PropTypes.number,
  children: PropTypes.element
};
export default App;
