// @dhis2/app-adapter/build/cjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _style = _interopRequireDefault(require("styled-jsx/style"));

var _ui = require("@dhis2/ui");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _Alerts = require("./components/Alerts");

var _AuthBoundary = require("./components/AuthBoundary");

var _FatalErrorBoundary = require("./components/FatalErrorBoundary");

var _ServerVersionProvider = require("./components/ServerVersionProvider");

var _styles = require("./styles.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const App = ({
  url,
  apiVersion,
  appName,
  children
}) => /*#__PURE__*/_react.default.createElement(_FatalErrorBoundary.FatalErrorBoundary, null, /*#__PURE__*/_react.default.createElement(_ServerVersionProvider.ServerVersionProvider, {
  url: url,
  apiVersion: apiVersion
}, /*#__PURE__*/_react.default.createElement("div", {
  className: "jsx-".concat(_styles.styles.__hash) + " " + "app-shell-adapter"
}, /*#__PURE__*/_react.default.createElement(_style.default, {
  id: _styles.styles.__hash
}, _styles.styles), /*#__PURE__*/_react.default.createElement(_ui.HeaderBar, {
  appName: appName
}), /*#__PURE__*/_react.default.createElement(_AuthBoundary.AuthBoundary, {
  url: url
}, /*#__PURE__*/_react.default.createElement("div", {
  className: "jsx-".concat(_styles.styles.__hash) + " " + "app-shell-app"
}, children)), /*#__PURE__*/_react.default.createElement(_Alerts.Alerts, null))));

App.propTypes = {
  appName: _propTypes.default.string.isRequired,
  url: _propTypes.default.string.isRequired,
  apiVersion: _propTypes.default.number,
  children: _propTypes.default.element
};
var _default = App;
exports.default = _default;
